src/Repositories/Eloquent/CategoryRepository.php

```diff
@@ -2,7 +2,62 @@
 
 namespace IndieHD\Velkart\Repositories\Eloquent;
 
-class CategoryRepository
+use Illuminate\Database\DatabaseManager;
+use IndieHD\Velkart\Models\Eloquent\Category;
+use IndieHD\Velkart\Contracts\CategoryRepositoryContract;
+use IndieHD\Velkart\Repositories\Eloquent\BaseRepository;
+
+class CategoryRepository extends BaseRepository implements CategoryRepositoryContract
 {
-    //
+    /**
+     * @var Category
+     */
+    protected $category;
+
+    /**
+     * @var DatabaseManager
+     */
+    protected $db;
+
+    /**
+     * CategoryRepository constructor.
+     * @param Category $category
+     * @param DatabaseManager $db
+     */
+    public function __construct(Category $category, DatabaseManager $db)
+    {
+        $this->category = $category;
+        $this->db = $db;
+    }
+
+    public function model(): Category
+    {
+        return $this->category;
+    }
+
+    public function modelClass(): string
+    {
+        return Category::class;
+    }
+
+    /**
+     * @param int $id
+     * @return bool
+     * @throws \Throwable
+     */
+    public function delete(int $id): bool
```

Comment:
I think the `BaseRepository` covers this logic. This repository probably doesn't need to overwrite this method ?

Reply:
My mistake this has been added in PR #16

Reply:
Suggest merging PR #16 and omitting this method as the logic doesn't exceed that of the base.